### ISSUE TITLE
Problems with Panel documentation

### ISSUE THREAD

**Issue by AQ18** [holoviz/panel#4184]
#### ALL software version info
Jupyter Lab 3.5.0, Panel 0.14.1, Param 1.12.2

### Context
I've been tinkering with HoloViews to make interactive plots from my physics simulations for a while, and was contemplating if introducing Param and Panel was the next step, so I started reading the Panel User Guide. It wasn't long before I was struggling to understand the guide, and after convincing myself that these were problems with the guide rather than my reading comprehension, I started documenting the issues I found.

I'm aware that #3869 is open and underway, so possibly nothing from the current guides will be carried through anyway. But people like me are reading the guides and getting confused right now, so I thought it was worth flagging to your attention.

# Issues
## APIs
* Reactive Functions section refers to Interact section as if Interact is above, but actually it is below.

## Parameters
* "Here even the `msg` button behaviour..." - there is no mention of a `msg` button so far.
* "As an example, we can map a string and a number Selector to a RadioButtonGroup and DiscretePlayer respectively." - the example also includes the AutocompleteInput widget, which isn't mentioned. We also switch from using the Row pane to the Param pane without any explanation of the latter.

## Pipelines
* The first indexed heading (Non-linear pipelines) is halfway down the page - shouldn't there be a heading nearer the top?
* class `Stage1` defines a `ready` parameter that is entirely superfluous and not referred to...?
* "To make sure that we don’t get a widget for c (as it will be set by the previous stage, not the user), we’ll set its precedence to be negative..." but the sample code
`c = param.Number(default=5, bounds=(0, None))`
does not set the precedence.
* "To display the pipeline we simply let it render itself:" The following code snippet is very confusing. Why have we reassigned `pipeline`? Why do we add an instance of Stage1 but the class of Stage 2?
```
pipeline = pn.pipeline.Pipeline(debug=True)
pipeline.add_stage('Stage 1', Stage1())
pipeline.add_stage('Stage 2', Stage2)

pipeline
```
* Non-linear pipelines: I don't think the 'dag' example works properly for me. Running the code and performing these exact steps:
1. Click the 'Result' node, prints "6"
2. Click the 'Add' node', prints "2 + 3"
3. Click the 'Result' node, prints "5"
4. 'Multiply' node, prints "2.000 * 3.000"
5. 'Result' node, prints "5" ???
6. 'Input' node, change Value2 to "4"
7. 'Multiply' node, prints "2.000 * 4.000"
8. 'Result' node, prints "5" ???
9. 'Add' node, prints "2 + 3" ???
10. 'Input' node, no change
11. 'Add' node, prints "2 + 4"
12. 'Result' node, prints "6"
13. 'Input' -> 'Multiply' -> 'Result', still prints "6"

As a complete novice I'm not sure what the intended behaviour is, but I doubt that 'Result' is supposed to ignore 'Multiply' forever.
* Programmatic flow control: A similar bug occurs as above. If I set the operator to 'Multiply', I can hit Go and get 6, then return to the first node and do that a few times. If I go back to the first node and change the operator to 'Add' before hitting Go, then Adding is all I'm allowed to do now - Result doesn't update when passing through the Multiply node.

This is where I stopped for now.

**Comment by droumis** [comment]
This is great, thanks for the feedback. In the future, it would be great if you could create a separate issue for a given docs page or section (e.g. 'Pipelines'). Additionally, please report bugs separately as well. 

- The pipeline bug is now in #4419 
- The pipeline docs issues will be resolved by #4422 

I think all that remains is addressing the APIs and Parameters docs

**Comment by droumis** [comment]
The APIs and Parameters user guides are now reworked into smaller How-to guides, and will be released along with Panel 1.0
